invisible_cities/core/fit_functions.py

```diff
@@ -4,16 +4,55 @@
 GML November 2016
 """
 
-import numpy  as np
-import pandas as pd
+import numpy   as np
+import pandas  as pd
+import inspect as insp
 import scipy.optimize
 import scipy.stats
 
+from functools import wraps
+
 from .                    import core_functions as coref
 from .  stat_functions    import poisson_sigma
 from .. evm.ic_containers import FitFunction
 
 
+def fixed_parameters(fn, **kwargs):
+    """
+    Wrapper to fix parameters of a function
+    which can then be used in curve_fit.
+    """
+    fn_sig    = insp.signature(fn)
+    fn_pars   = np.array(list(fn_sig.parameters))[1:]
+    par_gen   = (kwargs[k] if k in kwargs.keys() else np.nan for k in fn_pars)
```

Comment:
`kwargs` seems to behave like a dictionary, so you can also do `kwargs.get(k , np.nan)` instead of `kwargs[k] if k in kwargs.keys() else np.nan`